core/trino-main/src/main/java/io/trino/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -266,10 +268,7 @@ private TableHandle findTableScanHandle(PlanNode node)
                 return findTableScanHandle(((SemiJoinNode) node).getSource());
             }
             if (node instanceof JoinNode) {
-                JoinNode joinNode = (JoinNode) node;
-                if (joinNode.getType() == JoinNode.Type.INNER && isAtMostScalar(joinNode.getRight())) {
-                    return findTableScanHandle(joinNode.getLeft());
-                }
+                return findTableScanHandle(((JoinNode) node).getLeft());
```

Comment:
For "Ensure delete/update is applied on actual table specified in query" commit it's sufficient to add checkState/verify above.

Removing the `if (joinNode.getType() == JoinNode.Type.INNER && isAtMostScalar(joinNode.getRight()))` condition here seems to broaden the set of supported queries, and should be accompanied by test changes. 
Did you mean to make this change in "Support uncorrelated subquery in UPDATE assignments (SET)" commit?

(if moving changes, make sure to move `rewriteModifyTableScan` changes in sync, as they need to be symmetrical)